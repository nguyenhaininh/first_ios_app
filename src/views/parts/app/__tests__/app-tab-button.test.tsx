/*---------------------------------------------------------------------------------------------
 *  Copyright (c) Lunascape Corporation. All rights reserved.
 *--------------------------------------------------------------------------------------------*/
import * as React from 'react';
import AppTabButton from '../app-tab-button';

import * as renderer from 'react-test-renderer';

it('renders correctly when focused', () => {
  const onPress = jest.fn();
  const tree = renderer.create(
    <AppTabButton
      sourceOn={123}
      sourceOff={456}
      on={true}
      onPress={onPress} >
      ABC
    </AppTabButton>
  ).toJSON();
  expect(tree).toMatchSnapshot();
});

it('renders correctly when unfocsed', () => {
  const onPress = jest.fn();
  const tree = renderer.create(
    <AppTabButton
      sourceOn={123}
      sourceOff={456}
      on={false}
      onPress={onPress} >
      ABC
    </AppTabButton>
  ).toJSON();
  expect(tree).toMatchSnapshot();
});
