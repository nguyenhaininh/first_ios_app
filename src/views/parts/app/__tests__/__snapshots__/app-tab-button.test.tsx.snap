// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly when focused 1`] = `
<View
  accessible={true}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  style={
    Object {
      "alignItems": "center",
      "flex": 1,
      "height": "100%",
      "justifyContent": "center",
      "width": 62,
    }
  }
>
  <Image
    source={123}
    style={
      Object {
        "marginBottom": 2,
      }
    }
  />
  <Text
    accessible={true}
    allowFontScaling={true}
    ellipsizeMode="tail"
    style={
      Object {
        "color": "#004098",
        "fontSize": 9,
      }
    }
  >
    ABC
  </Text>
</View>
`;

exports[`renders correctly when unfocsed 1`] = `
<View
  accessible={true}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  style={
    Object {
      "alignItems": "center",
      "flex": 1,
      "height": "100%",
      "justifyContent": "center",
      "width": 62,
    }
  }
>
  <Image
    source={456}
    style={
      Object {
        "marginBottom": 2,
      }
    }
  />
  <Text
    accessible={true}
    allowFontScaling={true}
    ellipsizeMode="tail"
    style={
      Object {
        "color": "#4f5b77",
        "fontSize": 9,
      }
    }
  >
    ABC
  </Text>
</View>
`;
